src/sfdxFlags.ts

```diff
@@ -732,8 +737,10 @@ export function buildSfdxFlags(
       if (!isKeyOf(optionalBuiltinFlags, key)) {
         throw messages.createError('error.UnknownBuiltinFlagType', [key]);
       }
+      // @ts-ignore
       output[key] = optionalBuiltinFlags[key](flag);
     } else {
+      // @ts-ignore
```

Comment:
yeah, this one's harder because it could be almost anything.  Is there a way to use the types of flagsConfig entries to know what type you're actually sending to `validateCustomFlag`, and maybe use that in the return type of `buildSfdxFlags`.

ex (this totally doesn't work but something in this direction)

```
const output: Output & { [key: keyof FlagsConfig]: flags.Discriminated<flags.Option | flags.Boolean> } = {
```

another idea might be to try to derive the output type from FlagsConfig (which already has most of these properties plus `[key: string]: Optional<flags.Boolean | flags.Option | flags.Builtin>;`